From a9f642268c78ab4f28db64e09e966f76dfa80093 Mon Sep 17 00:00:00 2001
From: Hannes Winkler <hanneswinkler2000@web.de>
Date: Wed, 24 Jun 2026 13:51:16 -0400
Subject: [PATCH 2/8] always include debug symbols


diff --git a/engine/src/build/config/compiler/BUILD.gn b/engine/src/build/config/compiler/BUILD.gn
index 9edd083b5f..9c19b9cf85 100644
--- a/engine/src/build/config/compiler/BUILD.gn
+++ b/engine/src/build/config/compiler/BUILD.gn
@@ -1150,7 +1150,7 @@ config("symbols") {
       ldflags = [ "-gsource-map" ]
     }
   } else {
-    cflags = [ "-g2" ]
+    cflags = [ "-ggdb", "-fdebug-default-version=4" ]
     asmflags = [ "-g" ]
     if (is_apple) {
       swiftflags = [ "-g" ]
@@ -1163,14 +1163,14 @@ config("minimal_symbols") {
     # Linker symbols for backtraces only.
     ldflags = [ "/DEBUG" ]
   } else {
-    cflags = [ "-g1" ]
+    cflags = [ "-ggdb", "-fdebug-default-version=4" ]
     asmflags = [ "-g" ]
   }
 }
 
 config("no_symbols") {
   if (!is_win) {
-    cflags = [ "-g0" ]
+    cflags = [ "-ggdb", "-fdebug-default-version=4" ]
     asmflags = [ "-g0" ]
   }
 }
-- 
2.54.0

