From e843a5b2512657c227e4fc349969366021f2147b Mon Sep 17 00:00:00 2001
From: Hannes Winkler <hanneswinkler2000@web.de>
Date: Sun, 24 Mar 2024 19:26:18 +0000
Subject: [PATCH 4/8] build all gen_snapshots by default


diff --git a/engine/src/flutter/BUILD.gn b/engine/src/flutter/BUILD.gn
index 7a4d1eaab3..3b048fa761 100644
--- a/engine/src/flutter/BUILD.gn
+++ b/engine/src/flutter/BUILD.gn
@@ -103,6 +103,12 @@ group("flutter") {
   if (!is_fuchsia && !is_qnx) {
     public_deps += [ "//flutter/lib/snapshot:generate_snapshot_bins" ]
 
+    if (host_os == "linux") {
+      public_deps += [ "$dart_src/runtime/bin:gen_snapshot(//build/toolchain/linux:clang_x64)" ]
+      public_deps += [ "$dart_src/runtime/bin:gen_snapshot(//build/toolchain/linux:clang_arm64)" ]
+      public_deps += [ "$dart_src/runtime/bin:gen_snapshot(//build/toolchain/linux:clang_arm)" ]
+    }
+
     if (build_engine_artifacts) {
       public_deps += [
         "//flutter/build/dart:dart_sdk",
-- 
2.54.0

