From e7048375e7a1e3441e71e5d386e10462de44e724 Mon Sep 17 00:00:00 2001
From: Hannes Winkler <hanneswinkler2000@web.de>
Date: Wed, 24 Jun 2026 13:54:13 -0400
Subject: [PATCH 5/8] support cross-building linux engine from windows


diff --git a/engine/src/build/config/BUILDCONFIG.gn b/engine/src/build/config/BUILDCONFIG.gn
index ba687ba773..ffd159dc07 100644
--- a/engine/src/build/config/BUILDCONFIG.gn
+++ b/engine/src/build/config/BUILDCONFIG.gn
@@ -594,6 +594,9 @@ if (custom_toolchain != "") {
     }
   } else if (host_os == "mac") {
     host_toolchain = "//build/toolchain/mac:clang_$host_cpu"
+  } else if (host_os == "win") {
+    assert(is_clang)
+    host_toolchain = "//build/toolchain/win:clang_$host_cpu"
   }
   if (is_clang) {
     set_default_toolchain("//build/toolchain/linux:clang_$current_cpu")
diff --git a/engine/src/flutter/tools/gn b/engine/src/flutter/tools/gn
index b09708e23c..62d3e89f15 100755
--- a/engine/src/flutter/tools/gn
+++ b/engine/src/flutter/tools/gn
@@ -542,7 +542,7 @@ def to_gn_args(args):
   # does not exist. Further, we set the 'host_cpu' so that it shares the
   # bitwidth of the 32-bit arm target.
   if sys.platform.startswith(
-      ('cygwin', 'win')) and args.target_os == 'android' and gn_args['target_cpu'] == 'arm':
+      ('cygwin', 'win')) and (args.target_os == 'android' or args.target_os == 'linux') and gn_args['target_cpu'] == 'arm':
     gn_args['host_cpu'] = 'x86'
     gn_args['current_cpu'] = 'x86'
 
@@ -799,7 +799,7 @@ def to_gn_args(args):
   # There is a special case for Android on Windows because there we _only_ build
   # gen_snapshot, but the build defines otherwise make it look like the build is
   # for a host Windows build and make GN think we will be building ANGLE.
-  if is_host_build(args) or (args.target_os == 'android' and get_host_os() == 'win'):
+  if is_host_build(args) or ((args.target_os == 'android' or args.target_os == 'linux') and get_host_os() == 'win'):
     # Don't include git commit information.
     gn_args['angle_enable_commit_id'] = False
     # Do not build unnecessary parts of the ANGLE tree.
-- 
2.54.0

