From 7950985d65e78474960aa93eceea0627f2a20cc8 Mon Sep 17 00:00:00 2001
From: Hannes Winkler <hanneswinkler2000@web.de>
Date: Mon, 20 Jan 2025 16:23:24 +0000
Subject: [PATCH 7/8] don't use vulkan xcb/wayland WSI integrations


diff --git a/engine/src/build_overrides/vulkan_headers.gni b/engine/src/build_overrides/vulkan_headers.gni
index 8ef7d23936..bd1868b21b 100644
--- a/engine/src/build_overrides/vulkan_headers.gni
+++ b/engine/src/build_overrides/vulkan_headers.gni
@@ -6,6 +6,6 @@
 # set anything.
 
 if (is_linux) {
-  vulkan_use_x11 = !is_minimal_linux
-  vulkan_use_wayland = true
+  vulkan_use_x11 = false
+  vulkan_use_wayland = false
 }
-- 
2.54.0

